### PR TITLE
New malloc trace implementation for SapMachine21

### PR DESCRIPTION
**Pull request by schmelter-sap**
Downport from PR 1551 (head).

Needed fix for missing MUSL_LIBC define from build system.

fixes #1550

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: make/modules/java.base/lib/CoreLibraries.gmk [hunk 1]
```diff
@@ -209,8 +209,6 @@ endif
 # SapMachine 2023-11-28: build libmallochooks
 
 ifeq ($(call isTargetOs, linux macosx), true)
-  LIBJLI_SRC_DIRS := $(call FindSrcDirsForComponent, java.base, libmallochooks)
```
Comment (reinrich) [inline make/modules/java.base/lib/CoreLibraries.gmk]: Will you do the same in head?
https://github.com/SAP/SapMachine/blob/84a5dff6bdfbe7ba9885fb9891253869516c788c/make/modules/java.base/lib/CoreLibraries.gmk#L213
Might even be better to fix it there and backport the cleanup. As you prefer.

**Review by reinrich — APPROVED** [review]
I've compared the original commit and the backport. Looks good to me.
Cheers, Richard.

File: make/modules/java.base/lib/CoreLibraries.gmk (hunk 1 above)
Comment (schmelter-sap) [reply make/modules/java.base/lib/CoreLibraries.gmk]: Yes, that will be fixed in head as well.

Comment (RealCLanger) [reply make/modules/java.base/lib/CoreLibraries.gmk]: This will be part of https://github.com/SAP/SapMachine/pull/1638, the current PR for jdk-23+17

**Review by RealCLanger — APPROVED** [review]

**Pull request merged**